### DISCUSSION TITLE
How does Solid detect if a function is a component?

### DISCUSSION THREAD

**Discussion by shadowtime2000** [solidjs/solid#313] [Q&A] [answered]
Just wondering, how does Solid exactly detect if a function is a component? Is it the type signature kind of thing with it taking one argument and it returning JSX or is it with `render` being called on that component? How exactly does it figure out what to do its compilation on?

**Comment by ryansolid** [comment, marked as answer]
Components aren't special at all. They are simple functions that are being called with JSX syntax. It is JSX that I detect. I look for JSX and transform it. That is all the compiler does. Solid's approach is mostly a runtime reactivity. The optimal compilation in regards to the DOM and creating that reactive graph. By keeping components light I reduce additional overhead. I wish there was more to it than this but really components are just functions that execute once. And then they are gone.

**Reply by samuelstroschein** [reply]
It might be worthwhile to communicate the difference between Solid's "simple" reactive runtime and Solid's compiler that transforms JSX. 

Solid is presented as a compiler-based reactive framework similar to Svelte. However, the fact that Solid's reactivity does not require a compiler makes it more appealing. The "Introduction to Reactivity" [video](https://www.youtube.com/watch?v=J70HXl1KhWE) is impressive and intuitive. Communicating that a compiler is required to (solely?) transform JSX, and not to achieve reactivity, reduces the fear of relying on "compiler magic" and lock-in.

**Reply by biomassives** [reply]
video no longer up .. ;(
